go-controller/cmd/ovn-k8s-cni-overlay/ovn-k8s-cni-overlay.go

```diff
@@ -1,221 +1,29 @@
 package main
 
 import (
-	"encoding/json"
-	"fmt"
-	"net"
 	"os"
-	"strings"
-	"time"
 
-	"github.com/sirupsen/logrus"
-	"github.com/urfave/cli"
+	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/cni"
+	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/config"
 
 	"github.com/containernetworking/cni/pkg/skel"
 	"github.com/containernetworking/cni/pkg/types"
-	"github.com/containernetworking/cni/pkg/types/current"
 	"github.com/containernetworking/cni/pkg/version"
-	"k8s.io/apimachinery/pkg/api/resource"
-	"k8s.io/apimachinery/pkg/util/wait"
-
-	"github.com/openvswitch/ovn-kubernetes/go-controller/cmd/ovn-k8s-cni-overlay/app"
-	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/config"
-	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/kube"
-	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/urfave/cli"
 )
 
-func argString2Map(args string) (map[string]string, error) {
-	argsMap := make(map[string]string)
-
-	pairs := strings.Split(args, ";")
-	for _, pair := range pairs {
-		kv := strings.Split(pair, "=")
-		if len(kv) != 2 {
-			return nil, fmt.Errorf("ARGS: invalid pair %q", pair)
-		}
-		keyString := kv[0]
-		valueString := kv[1]
-		argsMap[keyString] = valueString
-	}
-
-	return argsMap, nil
-}
-
-var minRsrc = resource.MustParse("1k")
-var maxRsrc = resource.MustParse("1P")
-
-func validateBandwidthIsReasonable(rsrc *resource.Quantity) error {
-	if rsrc.Value() < minRsrc.Value() {
-		return fmt.Errorf("resource is unreasonably small (< 1kbit)")
-	}
-	if rsrc.Value() > maxRsrc.Value() {
-		return fmt.Errorf("resoruce is unreasonably large (> 1Pbit)")
-	}
-	return nil
-}
-
-func extractPodBandwidthResources(podAnnotations map[string]string) (int64, int64, error) {
-	ingress := int64(-1)
-	egress := int64(-1)
-	str, found := podAnnotations["kubernetes.io/ingress-bandwidth"]
-	if found {
-		ingressVal, err := resource.ParseQuantity(str)
-		if err != nil {
-			return -1, -1, err
-		}
-		if err := validateBandwidthIsReasonable(&ingressVal); err != nil {
-			return -1, -1, err
-		}
-		ingress = ingressVal.Value()
-	}
-	str, found = podAnnotations["kubernetes.io/egress-bandwidth"]
-	if found {
-		egressVal, err := resource.ParseQuantity(str)
-		if err != nil {
-			return -1, -1, err
-		}
-		if err := validateBandwidthIsReasonable(&egressVal); err != nil {
-			return -1, -1, err
-		}
-		egress = egressVal.Value()
-	}
-	return ingress, egress, nil
-}
-
-func initConfig(ctx *cli.Context, args *skel.CmdArgs) (*config.OVNNetConf, error) {
-	conf, err := config.ReadCNIConfig(args.StdinData)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load netconf: %v", err)
-	}
-
-	if _, err := config.InitConfigWithPath(ctx, conf.ConfigFilePath, &config.Defaults{
```

Comment:
It looks like we don't have any more users for InitConfigWithPath. So we should merge it in config.go with InitConfig()

Reply:
yep, this PR removes the need for the CNI overlay to use the config file.

Reply:
Still keeping this around for the feature to provide a non-default config file path through cli options. May be cleanup in another PR.